### PR TITLE
[4.0.0] Support building sample with local maven repository

### PR DESCRIPTION
**Pull request by YasasRangika**
## Purpose
To add support to building the sample locally and provide required guidance in README.

### CODE REVIEW

File: analytics-custom-data-provider/README.md [hunk 1]
```diff
@@ -3,29 +3,42 @@
...
 
 __Steps to add custom analytics data:__
 
 Add the necessary component versions (not less than the sample values provided in the existing pom file) for `carbon.apimgt.gateway.version`, `synapse.version` and `carbon.apimgt.common.analytics.version`in pom file in the root directory.
+<br>
+Since this feature support available in the latest update levels, you have to follow the below steps to add the required artifacts to the local m2 repository and point it as a repository in pom file.
+
+- Add `org.wso2.carbon.apimgt.gateway` and `org.wso2.carbon.apimgt.common.analytics` jars to the local m2 manually.
+
+      mvn install:install-file -Dfile=<PATH_TO_FILE>/org.wso2.carbon.apimgt.gateway_<COMPENENT_VERSION>.jar -DgroupId=org.wso2.carbon.apimgt -DartifactId=org.wso2.carbon.apimgt.gateway -Dversion=<COMPENENT_VERSION> -Dpackaging=jar
```
Comment (wasuradananjith) [inline analytics-custom-data-provider/README.md:23]:
```suggestion
      mvn install:install-file -Dfile=<PATH_TO_FILE>/org.wso2.carbon.apimgt.gateway_<COMPONENT_VERSION>.jar -DgroupId=org.wso2.carbon.apimgt -DartifactId=org.wso2.carbon.apimgt.gateway -Dversion=<COMPONENT_VERSION> -Dpackaging=jar
```

File: analytics-custom-data-provider/README.md [hunk 2]
```diff
@@ -3,29 +3,42 @@
...
+      mvn install:install-file -Dfile=<PATH_TO_FILE>/org.wso2.carbon.apimgt.common.analytics_<COMPENENT_VERSION>.jar -DgroupId=org.wso2.carbon.apimgt -DartifactId=org.wso2.carbon.apimgt.common.analytics -Dversion=<COMPENENT_VERSION> -Dpackaging=jar
```
Comment (wasuradananjith) [inline analytics-custom-data-provider/README.md:24]:
```suggestion
      mvn install:install-file -Dfile=<PATH_TO_FILE>/org.wso2.carbon.apimgt.common.analytics_<COMPONENT_VERSION>.jar -DgroupId=org.wso2.carbon.apimgt -DartifactId=org.wso2.carbon.apimgt.common.analytics -Dversion=<COMPONENT_VERSION> -Dpackaging=jar
```

File: analytics-custom-data-provider/README.md (hunk 1 above)
Comment (wasuradananjith) [inline analytics-custom-data-provider/README.md:23]:
```suggestion
      mvn install:install-file -Dfile=<PATH_TO_FILE>/org.wso2.carbon.apimgt.gateway_<COMPONENT_VERSION>.jar -DgroupId=org.wso2.carbon.apimgt -DartifactId=org.wso2.carbon.apimgt.gateway -Dversion=<COMPENENT_VERSION> -Dpackaging=jar
```

Comment (YasasRangika) [reply analytics-custom-data-provider/README.md]: fixed with https://github.com/wso2/samples-apim/pull/105/commits/b4bf649fb9851176cac452926a06b0b669929b12

thanks!

File: analytics-custom-data-provider/README.md (hunk 2 above)
Comment (YasasRangika) [reply analytics-custom-data-provider/README.md]: fixed with https://github.com/wso2/samples-apim/pull/105/commits/b4bf649fb9851176cac452926a06b0b669929b12

thanks!

File: analytics-custom-data-provider/README.md (hunk 1 above)
Comment (YasasRangika) [reply analytics-custom-data-provider/README.md]: fixed with https://github.com/wso2/samples-apim/pull/105/commits/b4bf649fb9851176cac452926a06b0b669929b12

thanks!

File: analytics-custom-data-provider/pom.xml [hunk 3]
```diff
@@ -105,10 +111,8 @@
         <maven.compiler.source>8</maven.compiler.source>
         <maven.compiler.target>8</maven.compiler.target>
 <!--        Below versions can be found by searching the component name in <APIM_HOME>/repository/components/plugins directory.-->
-<!--        org.wso2.carbon.apimgt.gateway component version-->
-        <carbon.apimgt.gateway.version>9.0.174.182</carbon.apimgt.gateway.version>
-<!--        org.wso2.carbon.apimgt.common.analytics component version-->
-        <carbon.apimgt.common.analytics.version>9.0.174.182</carbon.apimgt.common.analytics.version>
+<!--        org.wso2.carbon.apimgt component version-->
+        <carbon.apimgt.version>9.0.174.197</carbon.apimgt.version>
```
Comment (wasuradananjith) [inline analytics-custom-data-provider/pom.xml:115]: This version can change, right? Moving forward, this pack will contain higher minor component levels (greater than 197) each time the customer update to the latest U2 level. Hence, we might need to say something like, 9.0.174.<current_minor_component_level> for future customers, shouldn't we?

Comment (YasasRangika) [reply analytics-custom-data-provider/pom.xml]: +1 and fixed with https://github.com/wso2/samples-apim/pull/105/commits/db2e763493f7503269ea30dbb2bccb1b97612d43

thanks!

**Review by wasuradananjith — APPROVED** [review]

**Pull request merged**